sugar.go

```diff
@@ -0,0 +1,214 @@
+// ErrorLevel. It doesn't add to the message's structured context.
+func (s *SugaredLogger) Errorf(template string, args ...interface{}) {
+	if ce := s.core.Check(ErrorLevel, fmt.Sprintf(template, args...)); ce != nil {
+		ce.Write()
+	}
+}
+
+// DPanic logs a message and some key-value pairs using the underlying logger's
+// DPanic method. Keys and values are treated as they are in the With
+// method. (See Logger.DPanic for details.)
+func (s *SugaredLogger) DPanic(msg interface{}, keysAndValues ...interface{}) {
+	if ce := s.core.Check(DPanicLevel, sweetenMsg(msg)); ce != nil {
+		ce.Write(sweetenFields(keysAndValues, s.core)...)
+	}
+}
+
+// DPanicf uses fmt.Sprintf to construct a dynamic message, which is passed to
+// the underlying Logger's DPanic method. (See Logger.DPanic for details.) It
+// doesn't add to the message's structured context.
+func (s *SugaredLogger) DPanicf(template string, args ...interface{}) {
+	if ce := s.core.Check(DPanicLevel, fmt.Sprintf(template, args...)); ce != nil {
+		ce.Write()
+	}
+}
+
+// Panic logs a message and some key-value pairs at PanicLevel, then panics.
+// Keys and values are treated as they are in the With method.
+func (s *SugaredLogger) Panic(msg interface{}, keysAndValues ...interface{}) {
+	if ce := s.core.Check(PanicLevel, sweetenMsg(msg)); ce != nil {
+		ce.Write(sweetenFields(keysAndValues, s.core)...)
+	}
+}
+
+// Panicf uses fmt.Sprintf to construct a dynamic message and logs it at
+// PanicLevel, then panics. It doesn't add to the message's structured context.
+func (s *SugaredLogger) Panicf(template string, args ...interface{}) {
+	if ce := s.core.Check(PanicLevel, fmt.Sprintf(template, args...)); ce != nil {
+		ce.Write()
+	}
+}
+
+// Fatal logs a message and some key-value pairs at FatalLevel, then calls
+// os.Exit(1). Keys and values are treated as they are in the With method.
+func (s *SugaredLogger) Fatal(msg interface{}, keysAndValues ...interface{}) {
+	if ce := s.core.Check(FatalLevel, sweetenMsg(msg)); ce != nil {
+		ce.Write(sweetenFields(keysAndValues, s.core)...)
+	}
+}
+
+// Fatalf uses fmt.Sprintf to construct a dynamic message and logs it at
+// FatalLevel, then calls os.Exit(1). It doesn't add to the message's
+// structured context.
+func (s *SugaredLogger) Fatalf(template string, args ...interface{}) {
+	if ce := s.core.Check(FatalLevel, fmt.Sprintf(template, args...)); ce != nil {
+		ce.Write()
+	}
+}
+
+func sweetenFields(args []interface{}, errLogger Logger) []zapcore.Field {
+	if len(args) == 0 {
+		return nil
+	}
+	if len(args)%2 == 1 {
+		errLogger.DPanic(oddNumberErrMsg, Any("ignored", args[len(args)-1]))
+	}
+
+	fields := make([]zapcore.Field, len(args)/2)
+	for i := range fields {
+		key := sweetenMsg(args[2*i])
+		fields[i] = Any(key, args[2*i+1])
+	}
+	return fields
+}
+
+func sweetenMsg(msg interface{}) string {
+	if str, ok := msg.(string); ok {
+		return str
+	}
+	return fmt.Sprint(msg)
+}
```

Comment:
* I agree that a single allocation isn't a huge deal, but it is another drawback of this approach.
 * I'm not sure I agree that a `Stringer` is useful. Can you provide use cases from real world code where users are calling `.String()` and passing it as a message right now?
 * There is no single message that is passed as a `interface{}`.
   * logrus supports a list of `interface{}` that it then puts together, which is more similar to the standard library `log` package. This pattern is very different to the SugaredLogger, since a log line like `log.Info("dial failed", hostname, err)` will just append everything together with `logrus` or `log`, but with the `SugaredLogger`, it will treat `hostname` as a key. Considering how differently these work, I'm not sure it's a good comparison.
   * log15 expects messages to be strings, see the [Logger type](https://godoc.org/gopkg.in/inconshreveable/log15.v2#Logger).
   * The `go-kit/log` [Logger](https://godoc.org/github.com/go-kit/kit/log#Logger) doesn't even take a message, it only takes key-value pairs.